enumerables.rb

```diff
@@ -0,0 +1,96 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    elements.size.times do
+      yield(elements[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    size.times do
+      yield(elements[index], index)
+      index += 1
+    end
+  end
+
+  def my_select(&array)
+    return enum_for unless block_given?
+
+    selected_items = []
+    my_each do |x|
+      selected_items.push(x) unless array.call(x) == false
+    end
+    selected_items
+  end
+
+  def my_all?(array = nil)
+    if array
+      my_each { |x| return false unless array == x }
+    elsif !block_given?
+      my_each { |x| return false unless x }
+    else
+      my_each { |x| return false unless yield(x) }
+    end
+    true
+  end
+
+  def my_any?(args = nil)
+    if args
+      my_each { |x| return false unless args == x }
+    end
+    if !block_given?
+      my_each { |x| return true if x }
+    else
+      my_each { |x| return true if yield(x) }
+    end
+    false
+  end
+
+  def my_none?(args = nil, &block)
+    !my_any?(args, &block)
+  end
+
+  def my_count(arguement = nil)
+    counter = 0
+    if arguement
+      my_each { |x| counter += 1 if x == arguement }
+    elsif block_given?
+      my_each { |x| counter += 1 if yield(x) }
+    else
+      counter = length
+    end
+    counter
+  end
+
+  def my_map(proc = nil)
+    return to_enum unless block_given?
+
+    selected = []
+    if proc
+      my_each { |x| selected.push(proc.call(x)) }
+    else
+      my_each { |x| selected.push(yield(x)) }
+    end
+    selected
+  end
+
+  def my_inject(element = nil, symbol = nil &block)
+    element = element.to_sym if element.is_a?(String) && !symbol && !block
+    if element.is_a?(Symbol) && !symbol
+      block = element.to_proc
+      element = nil
+    end
+    symbol = symbol.to_sym if symbol.is_a?(String)
+    block = symbol.to_proc if symbol.is_a?(Symbol)
+    my_each { |x| element = element.nil? ? x : block.yield(element, x) }
+    element
+  end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Yours is retrieving a NameError
![image](https://user-images.githubusercontent.com/52209067/80156175-83194180-8580-11ea-87a5-a9366a856554.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(:+) == [2, 5, 6].inject(:+) => 13
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/80142486-74bf2b80-8568-11ea-9bdb-ca526cf0ee6b.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698534-6a643e80-6f76-11ea-9d0e-ebb0f02da179.png)